consumer.go

```diff
@@ -1004,6 +1009,8 @@ func (r *Consumer) redistributeRDY() {
 		availableMaxInFlight = 1 - atomic.LoadInt64(&r.totalRdyCount)
 	}
 
+	r.rngMtx.Lock()
+	defer r.rngMtx.Unlock()
```

Comment:
can we reduce the size of this critical path and put this lock around _just_ the call to `r.rng.Int()`?